### PR TITLE
Increase Titan default max-background-gc to 4

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Signed-off-by: Yi Wu <yiwu@pingcap.com>

###  What have you changed?
Current default of titan max-background-gc is 1, which is too small. Increase it to 4, or num_cpu, whichever is smaller.

###  What is the type of the changes?
Improvement

###  How is the PR tested?
start a tikv with titan enabled manually. check rocksdb log to make sure max-background-gc=4.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
no

###  Does this PR affect `tidb-ansible`?
will update config template.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -670,7 +674,7 @@ impl Default for TitanDBConfig {
             enabled: false,
             dirname: "".to_owned(),
             disable_gc: false,
-            max_background_gc: 1,
+            max_background_gc: 4,
```
Comment (zhangjinpeng1987) [inline src/config.rs]: Do we need to adjust it by the number of CPU cores?

File: etc/config-template.toml [hunk 2]
```diff
@@ -462,8 +462,8 @@
 # enabled = false
 
 ## Maximum number of threads of `Titan` background gc jobs.
-# default: 1
-# max-background-gc = 1
+# default: 4
+# max-background-gc = 4
```
Comment (zhangjinpeng1987) [inline etc/config-template.toml]: What is the difference between `default` and `max-background-gc`?

Comment (yiwu-arbug) [reply etc/config-template.toml]: `default` is just a comment. `max-background-gc` is which user can un-comment and make change.

**Comment by yiwu-arbug** [comment]
> is that too large which may influence foreground work?

IMO limiting the thread pool is just to limit context switch. To limit background work user should set rate-bytes-per-sec.

**Pull request merged**